catalyst_voices/apps/voices/lib/pages/proposal/proposal_navigation_panel.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:catalyst_voices/common/ext/build_context_ext.dart';
+import 'package:catalyst_voices/widgets/widgets.dart';
+import 'package:catalyst_voices_assets/catalyst_voices_assets.dart';
+import 'package:catalyst_voices_blocs/catalyst_voices_blocs.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_view_models/catalyst_voices_view_models.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class ProposalNavigationPanel extends StatelessWidget {
+  const ProposalNavigationPanel({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: SegmentsControllerScope.of(context),
+      builder: (context, value, _) {
+        return _ProposalNavigationPanel(
+          segments: value.segments,
+          openedSegments: value.openedSegments,
+          selectedSectionId: value.activeSectionId,
+        );
+      },
+    );
+  }
+}
+
+class _ProposalNavigationPanel extends StatelessWidget {
+  final List<Segment> segments;
+  final Set<NodeId> openedSegments;
+  final NodeId? selectedSectionId;
+
+  const _ProposalNavigationPanel({
+    required this.segments,
+    this.openedSegments = const {},
+    this.selectedSectionId,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView.separated(
+      padding: const EdgeInsets.symmetric(vertical: 16),
+      itemBuilder: (context, index) {
+        if (index == 0) {
+          return const _ControlsTile(key: ValueKey('NavigationControls'));
+        }
+        index--;
+
+        final segment = segments[index];
```

Comment:
I think more often we just write it like this instead of decrementing:

```suggestion
        final segment = segments[index - 1];
```